GNU assembler does not recognize `.dword` in `libafl_qemu.h`

**IMPORTANT**
1. You have verified that the issue to be present in the current `main` branch

Thank you for making LibAFL better!

**Describe the bug**
I'm working on a fuzzer under x86-64 using sync exits provided by qemu system mode. I use GCC 11.4.0 to compile the target. It seems that GNU assembler does not recognize the directive `.dword` in the header `libafl_qemu.h`. It can be easily fixed by replacing `.dword` with `.4byte` or `.long`.

**To Reproduce**
Steps to reproduce the behavior:
1. Include `libafl_qemu.h` in some source files and compile with GCC.

**Expected behavior**
The output is like:
```
test.c: Assembler messages:
test.c:10: Error: unknown pseudo-op: `.dword'
```

--- a/libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs
+++ b/libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs
@@ -236,8 +236,8 @@ pub fn merge(
                     .scheduler_mut()
                     .on_remove(&mut state, idx, &Some(testcase))?;
             } else {
-                #[allow(clippy::needless_borrows_for_generic_args)]
                 // False-positive: file_path is used just below
+                #[allow(clippy::needless_borrows_for_generic_args)]
                 rename(&file_path, &new_file_path)?;
                 *file_path = new_file_path;
             }

--- a/libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs
+++ b/libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs
@@ -236,7 +236,8 @@ pub fn merge(
                     .scheduler_mut()
                     .on_remove(&mut state, idx, &Some(testcase))?;
             } else {
-                #[allow(clippy::needless_borrows_for_generic_args)] // False-positive: file_path is used just below
+                #[allow(clippy::needless_borrows_for_generic_args)]
+                // False-positive: file_path is used just below
                 rename(&file_path, &new_file_path)?;
                 *file_path = new_file_path;
             }

--- a/libafl_qemu/Cargo.toml
+++ b/libafl_qemu/Cargo.toml
@@ -100,6 +100,7 @@ libafl_qemu_build = { path = "./libafl_qemu_build", version = "0.12.0" }
 pyo3-build-config = { version = "0.21", optional = true }
 rustversion = "1.0"
 bindgen = "0.69"
+cc = "1.0"
 
 [lib]
 name = "libafl_qemu"

--- a/libafl_qemu/build_linux.rs
+++ b/libafl_qemu/build_linux.rs
@@ -6,6 +6,13 @@ use std::{
 
 use libafl_qemu_build::maybe_generate_stub_bindings;
 
+static LIBAFL_QEMU_RUNTIME_TEST: &str = r#"
+#include <stdio.h>
+#include "libafl_qemu.h"
+
+int main() {}
+"#;
+
 #[allow(clippy::too_many_lines)]
 pub fn build() {
     // Note: Unique features are checked in libafl_qemu_sys
@@ -41,6 +48,17 @@ pub fn build() {
 
     let libafl_runtime_dir = src_dir.join("runtime");
     let libafl_qemu_hdr = libafl_runtime_dir.join(libafl_qemu_hdr_name);
+
+    let libafl_runtime_testfile = out_dir.join("runtime_test.c");
+    fs::write(&libafl_runtime_testfile, LIBAFL_QEMU_RUNTIME_TEST).expect("Could not write runtime test file");
+
+    let mut runtime_test_cc_compiler = cc::Build::new();
+
+    runtime_test_cc_compiler.cpp(false)
+        .include(&libafl_runtime_dir)
+        .file(&libafl_runtime_testfile);
+
+    runtime_test_cc_compiler.try_compile("runtime_test").unwrap();
 
     let runtime_bindings_file = out_dir.join("libafl_qemu_bindings.rs");
     let stub_runtime_bindings_file = src_dir.join("runtime/libafl_qemu_stub_bindings.rs");

--- a/libafl_qemu/runtime/libafl_qemu.h
+++ b/libafl_qemu/runtime/libafl_qemu.h
@@ -87,7 +87,7 @@ typedef enum LibaflQemuEndStatus {
         libafl_word ret;                                                                            \
         __asm__ volatile (                                                                        \
               "mov %1, %%rax\n"                                                                   \
-              ".dword " XSTRINGIFY(opcode) "\n"                                                   \
+              ".4byte " XSTRINGIFY(opcode) "\n"                                                   \
               "mov %%rax, %0\n"                                                                   \
               : "=g"(ret)                                                                         \
               : "g"(action)                                                                       \
@@ -102,7 +102,7 @@ typedef enum LibaflQemuEndStatus {
         __asm__ volatile (                                                                        \
             "mov %1, %%rax\n"                                                                     \
             "mov %2, %%rdi\n"                                                                     \
-            ".dword " XSTRINGIFY(opcode) "\n"                                                     \
+            ".4byte " XSTRINGIFY(opcode) "\n"                                                     \
             "mov %%rax, %0\n"                                                                     \
             : "=g"(ret)                                                                           \
             : "g"(action), "g"(arg1)                                                              \
@@ -118,7 +118,7 @@ typedef enum LibaflQemuEndStatus {
         "mov %1, %%rax\n"                                                                         \
         "mov %2, %%rdi\n"                                                                         \
         "mov %3, %%rsi\n"                                                                         \
-        ".dword " XSTRINGIFY(opcode) "\n"                                                         \
+        ".4byte " XSTRINGIFY(opcode) "\n"                                                         \
         "mov %%rax, %0\n"                                                                         \
         : "=g"(ret)                                                                               \
         : "g"(action), "g"(arg1), "g"(arg2)                                                       \

